tensornetwork/matrixproductstates/mps.py

```diff
@@ -341,6 +431,315 @@ def measure_two_body_correlator(self, op1: Union[BaseNode, Tensor],
           L = self.apply_transfer_operator(n % N, 'left', L)
     return c
 
+  def apply_two_site_gate(self,
+                          gate: Union[BaseNode, Tensor],
+                          site1: int,
+                          site2: int,
+                          max_singular_values: Optional[int] = None,
+                          max_truncation_err: Optional[float] = None) -> Tensor:
+    """
+    Apply a two-site gate to an MPS. This routine will in general 
+    destroy any canonical form of the state. If a canonical form is needed, 
+    the user can restore it using `FiniteMPS.position`.
+
+    Args:
+      gate (Tensor): a two-body gate
+      site1, site2 (int, int): the sites where the gate should be applied
+      max_singular_values (int): The maximum number of singular values to keep.
+      max_truncation_err (float): The maximum allowed truncation error.
+    Returns:
+      scalar `Tensor`: the truncated weight of the truncation.
+    
+    """
+    if len(gate.shape) != 4:
+      raise ValueError('rank of gate is {} but has to be 4'.format(
+          len(gate.shape)))
+    if site1 < 0 or site1 >= len(self) - 1:
+      raise ValueError(
+          'site1 = {} is not between 0 <= site < N - 1 = {}'.format(
+              site1, len(self)))
+    if site2 < 1 or site2 >= len(self):
+      raise ValueError('site2 = {} is not between 1 <= site < N = {}'.format(
+          site2, len(self)))
+    if site2 <= site1:
+      raise ValueError('site2 = {} has to be larger than site2 = {}'.format(
+          site2, site1))
+    if site2 != site1 + 1:
+      raise ValueError(
+          'site2 ={} != site1={}. Only nearest neighbor gates are currently '
+          'supported'.format(site2, site1))
```

Comment:
These two checks seem redundant.

Reply:
The second check is there to inform users that currently only nearest neighbor gates are supported. This may change in the future.